src/test/java/teammates/test/cases/action/InstructorFeedbackTemplateQuestionAddActionTest.java

```diff
@@ -0,0 +1,211 @@
+        FeedbackSessionAttributes fs = typicalBundle.feedbackSessions.get("session1InCourse1");
+
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, fs.getCourseId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, fs.getFeedbackSessionName(),
+                Const.ParamsNames.FEEDBACK_QUESTION_TEMPLATE_NUMBER, "2"
+        };
+        params = addUserIdToParams(instructor1ofCourse1.googleId, params);
+
+        InstructorFeedbackTemplateQuestionAddAction action = getAction(params);
+        RedirectResult result = getRedirectResult(action);
+
+        assertEquals(
+                getPageResultDestination(
+                        Const.ActionURIs.INSTRUCTOR_FEEDBACK_EDIT_PAGE,
+                        instructor1ofCourse1.courseId,
+                        "First+feedback+session",
+                        instructor1ofCourse1.googleId,
+                        false),
+                result.getDestinationWithParams());
+
+        String expectedLogMessage = "TEAMMATESLOG|||instructorFeedbackTemplateQuestionAdd|||"
+                                    + "instructorFeedbackTemplateQuestionAdd|||true|||"
+                                    + "Instructor(M)|||Instructor 1 of Course 1|||"
+                                    + "idOfInstructor1OfCourse1|||instr1@course1.tmt|||"
+                                    + "Added Feedback Template Question for Feedback Session:<span class=\"bold\">"
+                                    + "(First feedback session)</span> for Course "
+                                    + "<span class=\"bold\">[idOfTypicalCourse1]</span>"
+                                    + " created.<br><span class=\"bold\">Essay question:</span> "
+                                    + "Comments about your contribution (shown to other teammates)|||"
+                                    + "/page/instructorFeedbackTemplateQuestionAdd";
+        AssertHelper.assertLogMessageEqualsInMasqueradeMode(expectedLogMessage, action.getLogMessage(), adminUserId);
+
+        FeedbackQuestionAttributes fqa = FeedbackQuestionsLogic.inst().getFeedbackQuestion(fs.getFeedbackSessionName(),
+                fs.getCourseId(), 6);
+        assertNotNull(fqa);
+    }
+
+    protected String getPageResultDestination(
+            String parentUri, String courseId, String fsname, String userId, boolean isError) {
+        String pageDestination = parentUri;
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.COURSE_ID, courseId);
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.FEEDBACK_SESSION_NAME, fsname);
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.USER_ID, userId);
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.ERROR, Boolean.toString(isError));
+        return pageDestination;
+    }
+
+    @Override
+    @Test
+    public void testAccessControl() {
+        String[] params = new String[] {
```

Comment:
The `params` should be valid here? (i.e. should have `FEEDBACK_QUESTION_TEMPLATE_NUMBER` param?)